### PR TITLE
Add `op_is_terminal` to deno_core

### PR DESCRIPTION
**Pull request by nathanwhit**

### CODE REVIEW

File: testing/checkin/runner/ops_io.rs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+use deno_core::anyhow::Error;
```
Comment (mmastrac) [inline testing/checkin/runner/ops_io.rs:1]: This bites me all the time -- run `tools/lint.ts --fix` :)

Comment (bartlomieju) [reply testing/checkin/runner/ops_io.rs]: @dsherret suggested to put an empty line after copyright and that will make rust-analyzer behave

File: testing/checkin/runner/ops_io.rs [hunk 2]
```diff
@@ -65,3 +69,41 @@ pub fn op_pipe_create(op_state: &mut OpState) -> (ResourceId, ResourceId) {
...
+
+impl FileResource {
+  fn new(file: tokio::fs::File) -> Self {
+    let fd = file.as_raw_fd();
+    Self { fd }
+  }
+}
+
+impl Resource for FileResource {
+  fn backing_fd(
```
Comment (mmastrac) [inline testing/checkin/runner/ops_io.rs:85]: I think you'll want to override this one instead to make it work on unix and windows:

```
  fn backing_handle(self: Rc<Self>) -> Option<ResourceHandle> {

  }
```

File: testing/unit/resource_test.ts [hunk 3]
```diff
@@ -47,3 +47,8 @@ test(function opsSyncBadResource() {
     assert(e instanceof Deno.core.BadResource);
   }
 });
+
+test(async function testFileIsNotTerminal() {
```
Comment (mmastrac) [inline testing/unit/resource_test.ts:51]: Looks great!

**Review by mmastrac — APPROVED** [review]
LGTM!

**Pull request merged**